pkg/index/index_factory.go

```diff
@@ -0,0 +1,84 @@
+package index
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/buildpacks/imgutil"
+	"github.com/buildpacks/imgutil/layout"
+	"github.com/buildpacks/imgutil/remote"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/pkg/errors"
+)
+
+type IndexFactory struct {
```

Comment:
I think we need to test coverage here, I will check from Sai's branch to restore some of them

Reply:
This is still missing I think